### PR TITLE
Bug fixes for issues found in testing queues

### PR DESCRIPTION
**Pull request by mikalstill**
Make sure we don't infinitely attempt to place instances. Fix a queue reporting bug. Don't infinitely attempt to delete instances.

### CODE REVIEW

File: shakenfist/db.py [hunk 1]
```diff
@@ -238,6 +239,7 @@ def place_instance(instance_uuid, node):
         return
```
Comment (mandoonandy) [inline shakenfist/db.py:239]: Does this check block the recording of a placement attempt? If not, maybe it shouldn't be called?

**Review by mandoonandy — COMMENTED** [review]
I'm unclear (without more investigation) how the instance start queuing works.

_preflight and _start are queued at the same time, but _preflight can be requeued? Does that put the _preflight after the _start?

File: shakenfist/db.py (hunk 1 above)
Comment (mikalstill) [reply shakenfist/db.py:239]: So, a new attempt shouldn't be allowed to use the current node, so it shouldn't? That said, there might be a bug in the caller then?

File: shakenfist/daemons/queues.py [hunk 2]
```diff
@@ -103,6 +108,10 @@ def instance_preflight(instance_uuid, network):
                      'schedule', 'retry', None,
                      'insufficient resources: ' + str(e))
 
+    if instance.db_entry.get('placement_attempts') > 3:
+        raise exceptions.AbortInstanceStartException(
+            'Too many start attempts.')
+
     try:
         if instance.db_entry.get('requested_placement'):
             candidates = [instance.db_entry.get('requested_placement')]
```
Comment (mikalstill) [inline shakenfist/daemons/queues.py:117]: Yes, there is a bug in this logic. The list of candidate should exclude the current node. I will fix.

**Comment by mikalstill** [comment]
Sorry, I missed your greater comment earlier. The preflight is a re-check that the node is still capable of scheduling the instance. If its not, the instance is re-scheduled on another node and the entire workitem should be re-queued there. Does that make sense?

File: shakenfist/util.py [hunk 3]
```diff
@@ -260,3 +261,22 @@ def ignore_exception(processname, e):
         msg += '\n%s' % traceback.format_exc()
 
     LOG.error(msg)
+
+
+METHOD_RE = re.compile('.* in (.*)')
+
+
+def log(level, relatedobjects, message):
```
Comment (mandoonandy) [inline shakenfist/util.py:269]: I think you should have a command for each log level.

**Review by mandoonandy — COMMENTED** [review]
My general comment is that if we go down the path of high information logging, then we should do it properly with parameterised fields. Then it can be ingested easily into any of the log processors.
It will also be less verbose then using various ( [ : to indicate a field.

File: shakenfist/util.py (hunk 3 above)
Comment (mikalstill) [reply shakenfist/util.py:269]: Mutter.

**Comment by mikalstill** [comment]
Tweaks based on our slack conversation.

**Comment by mikalstill** [comment]
Logs look like:

```
Sep 10 10:56:18 cbr-sf-3 INFO sf-queues[92312] Processing workitem; method=daemons/queues.py:29:handle(); workitem=1599735378.7058043-000
Sep 10 10:56:18 cbr-sf-3 INFO sf-queues-1599735378.7058043-000[92312] Executing task instance_preflight; instance=ebbc70c7-ce6e-4eb2-8371-4d9313011d23; method=daemons/queues.py:51:handle(); workitem=1599735378.7058043-000
Sep 10 10:56:18 cbr-sf-3 INFO sf-queues-1599735378.7058043-000[92312] Executing task image_fetch; instance=ebbc70c7-ce6e-4eb2-8371-4d9313011d23; method=daemons/queues.py:51:handle(); workitem=1599735378.7058043-000
Sep 10 10:56:18 cbr-sf-2 INFO sf-queues-1599735378.4955704-000[2050934] Executing task image_fetch; instance=64d1f8d9-4b3b-486b-8422-e25c9af28147; method=daemons/queues.py:51:handle(); workitem=1599735378.4955704-000
```

**Review by mandoonandy — APPROVED** [review]

**Pull request merged**